### PR TITLE
:arrow_up: Upgrade typos and its configuration

### LINKED ISSUE #16610: Upgrade typos job in CI and fix the typos detected
The new `typos` version catches a bunch of new typos. We should fix them, and then update.

_Originally posted by @alice-i-cecile in https://github.com/bevyengine/bevy/issues/16602#issuecomment-2511825030_

### PR DESCRIPTION
**Pull request by homersimpsons**
# Objective

Fixes #16610, related to #16702

## Solution

Upgrade typos and its configuration

## Testing

- Did you test these changes? If so, how? No
- Are there any parts that need more testing? No
- How can other people (reviewers) test your changes? Is there anything specific they need to know? No
- If relevant, what platforms did you test these changes on, and are there any important ones you can't test? Not applicable

### CODE REVIEW

File: crates/bevy_ui/src/stack.rs [hunk 1]
```diff
@@ -35,7 +35,7 @@ impl ChildBufferCache {
 
 /// Generates the render stack for UI nodes.
 ///
-/// Create a list of root nodes from unparented entities and entities with a `GlobalZIndex` component.
+/// Create a list of root nodes from parentless entities and entities with a `GlobalZIndex` component.
```
Comment (homersimpsons) [inline crates/bevy_ui/src/stack.rs:38]: We are using `parentless` in similar contexts:

https://github.com/bevyengine/bevy/blob/0707c0717b6a3156a890d8800b84d746a6c29bfd/crates/bevy_transform/src/commands.rs#L89

https://github.com/bevyengine/bevy/blob/0707c0717b6a3156a890d8800b84d746a6c29bfd/crates/bevy_ui/src/layout/ui_surface.rs#L19-L20

File: crates/bevy_mesh/src/index.rs [hunk 2]
```diff
@@ -37,7 +37,7 @@ where
 #[derive(Debug, Error)]
 pub enum MeshWindingInvertError {
     /// This error occurs when you try to invert the winding for a mesh with [`PrimitiveTopology::PointList`](super::PrimitiveTopology::PointList).
-    #[error("Mesh winding invertation does not work for primitive topology `PointList`")]
+    #[error("Mesh winding inversion does not work for primitive topology `PointList`")]
```
Comment (homersimpsons) [inline crates/bevy_mesh/src/index.rs:40]: Suggested in https://github.com/bevyengine/bevy/pull/16702#issuecomment-2525354642

File: typos.toml [hunk 3]
```diff
@@ -7,32 +7,27 @@ extend-exclude = [
...
-
 # Match Inside a Word - Case Insensitive
 [default.extend-words]
 LOD = "LOD" # Level of detail
 TOI = "TOI" # Time of impact
 
 [default]
 locale = "en-us"
+# Ignored typos regexes
 extend-ignore-identifiers-re = [
```
Comment (homersimpsons) [inline typos.toml:18]: I believe it is easier to have everything in the same place. If I understand correctly, `default.extend-identifiers` is better used to define replacements or deny some usages.

File: typos.toml [hunk 4]
```diff
@@ -7,32 +7,27 @@ extend-exclude = [
...
+  "ser",              # ron::ser - Serializer
+  "SME",              # Subject Matter Expert
+  "Sur",              # macOS Big Sur - South
+  "NDK",              # NDK - Native  Development Kit
+  "PNG",              # PNG - Portable Network Graphics file format
+  "Masia",            # The surname of one of the authors of SMAA
+  "metalness",        # Rendering term (metallicity)
+  "inventario",       # Inventory in Portuguese
+  "hemicircles",      # Geometry term (semicircle)
+  "[Rr]eparametrize", # Mathematical term in curve context (reparameterize)
```
Comment (homersimpsons) [inline typos.toml:30]: They are ordered by length, maybe some other orders could be more relevant?

File: typos.toml [hunk 5]
```diff
@@ -7,32 +7,27 @@ extend-exclude = [
...
+  "ba",               # Part of an accessor in WGSL - color.ba
+  "ser",              # ron::ser - Serializer
+  "SME",              # Subject Matter Expert
+  "Sur",              # macOS Big Sur - South
+  "NDK",              # NDK - Native  Development Kit
+  "PNG",              # PNG - Portable Network Graphics file format
+  "Masia",            # The surname of one of the authors of SMAA
+  "metalness",        # Rendering term (metallicity)
+  "inventario",       # Inventory in Portuguese
+  "hemicircles",      # Geometry term (semicircle)
```
Comment (BenjaminBrienen) [inline typos.toml:29]: I think hemicircles is en-UK

**Review by BenjaminBrienen — CHANGES_REQUESTED** [review]

File: typos.toml (hunk 5 above)
Comment (alice-i-cecile) [reply typos.toml:29]: Yeah, we should swap to semicircles if possible.

**Review by alice-i-cecile — APPROVED** [review]

File: typos.toml (hunk 5 above)
Comment (homersimpsons) [reply typos.toml:29]: It is used only twice in comments if I recall correctly. I can update it in this pull request.

Comment (alice-i-cecile) [reply typos.toml:29]: That would be great :) I won't block merging on that though.

Comment (homersimpsons) [reply typos.toml]: Updated

**Review by BenjaminBrienen — APPROVED** [review]

**Comment by BenjaminBrienen** [comment]
My own typos PR is becoming less and less relevant lol

**Pull request merged**